lib/truffle/truffle/cext.rb

```diff
@@ -1916,7 +1916,13 @@ def rb_enc_from_encoding(rb_encoding)
   end
 
   def RSTRING_PTR(string)
-    Truffle.invoke_primitive(:string_get_rstring_ptr, string)
+    ptr = Truffle.invoke_primitive(:string_get_rstring_ptr, string)
+    unless ptr
+      Truffle.privately do
+        ptr = RStringPtr.new(string)
```

Comment:
As far as I can tell, the `string_set_rstring_ptr` primitive is never called and needs to be.

Reply:
Good catch, I think I did a `Ctrl-Z` too much.